backend/data-store/db-client/store.go

```diff
@@ -64,6 +97,325 @@ func Open(ctx context.Context, log *zap.Logger) (*Store, error) {
 	return store, nil
 }
 
+func (st *Store) GetSubmissionDetails(ctx context.Context, submissionId int32) (*pb.DetailsSubmissionResponse, error) {
+	details, err := queries.New(st.Pool).GetSubmissionDetails(ctx, submissionId)
+	if err == pgx.ErrNoRows {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("getting submission details: %w", err)
+	}
+
+	budget, err := details.Budget.EncodeText(nil, []byte{})
+
+	if err != nil { // is it normal that column is in undefined state?
+		budget = []byte{} // hope so
+	}
+
+	var submissionDate string
+	if details.ReportSubmissionDate.Valid && details.ReportSubmissionDate.Time != (time.Time{}) {
```

Comment:
I don't get why would we return empty string instead of `nil`, but maybe this is the way to go in GoLang/gRPC. Anyway, sticking to the convention, I think this code should reside in a separate `DenullifyTime` method.